src/tracing/perfetto/perfetto_tracing.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <tracing/perfetto/perfetto_tracing.h>
+
+#include <lib/address_resolve/TracingStructs.h>
+#include <lib/support/ErrorStr.h>
+#include <lib/support/StringBuilder.h>
+#include <transport/TracingStructs.h>
+
+#include <matter/tracing/macros_impl.h>
+#include <perfetto.h>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
```

Comment:
Can we separate this from the configuration code below?

The application should be able to configure perfetto itself, but we need to define our own categories regardless. So everyone needs this line, while the code below probably belongs in examples/.

It really confuses matter to mash this altogether. We should keep in mind that on Android, the application does not initiate the trace - the system service manages that. All of this open/close code, or file backing, should not be included in those builds.

Reply:
File backing is currently controller by the input arg.

Android likely requires separate flags/logic, for now I just started with linux support. I am trying to be able to build things incrementally.

Reply:
If an app needs a custom tracing, I would imagine then they would not use this file at all. 

My thinking is that this is a perfetto implementation that one could use, however others could be used as well, including some `modified perfetto`

Reply:
Separated things out into:
  - separate track event registration (in case some apps will want to register more categories somehow)
  - file output config
  - perfetto startup

In theory applications can do their own config and not use any of these configurations. Eventually we could move them to "examples".